azure/src/main/scala/quasar/blobstore/azure/requests.scala

```diff
@@ -16,106 +16,91 @@
 
   final case class ListBlobHierarchyArgs(
-    containerURL: ContainerURL,
-    marker: Option[String],
+    containerClient: BlobContainerAsyncClient,
     delimiter: String,
-    options: ListBlobsOptions,
-    context: Context)
-
-  @SuppressWarnings(Array("org.wartremover.warts.Null"))
-  def listRequest[F[_]: Async: ContextShift](
-    args: ListBlobHierarchyArgs): F[ContainerListBlobHierarchySegmentResponse] =
-    Sync[F].delay(
-      args.containerURL.listBlobsHierarchySegment(
-        args.marker.orNull,
-        args.delimiter,
-        args.options,
-        args.context)) >>=
-      rx.singleToAsync[F, ContainerListBlobHierarchySegmentResponse]
-
-  def listRequestK[F[_]: Async: ContextShift]
-      : Kleisli[F, ListBlobHierarchyArgs, ContainerListBlobHierarchySegmentResponse] =
+    options: ListBlobsOptions)
+
+  def listRequest[F[_]: ConcurrentEffect: ContextShift](
+      args: ListBlobHierarchyArgs)
+    : F[Stream[F, BlobItem]] =
+      Sync[F].delay(
+        reactive.fluxToStream[F, BlobItem](
+          args.containerClient.listBlobsByHierarchy(
+            args.delimiter,
+            args.options)))
+
+  def listRequestK[F[_]: ConcurrentEffect: ContextShift]
+      : Kleisli[F, ListBlobHierarchyArgs, Stream[F, BlobItem]] =
     Kleisli(listRequest[F])
 
   final case class BlobPropsArgs(
-    blobURL: BlobURL,
-    blobAccessConditions: BlobAccessConditions,
-    context: Context)
+    blobClient: BlobAsyncClient,
+    blobRequestConditions: BlobRequestConditions)
 
   def blobPropsRequest[F[_]: Async: ContextShift](
-    args: BlobPropsArgs): F[BlobGetPropertiesResponse] =
-    Sync[F].delay(args.blobURL.getProperties(args.blobAccessConditions, args.context)) >>=
-      rx.singleToAsync[F, BlobGetPropertiesResponse]
+      args: BlobPropsArgs)
+      : F[Response[BlobProperties]] =
+    Sync[F].delay(args.blobClient.getPropertiesWithResponse(args.blobRequestConditions)) >>=
+      reactive.monoToAsync[F, Response[BlobProperties]]
 
   def blobPropsRequestK[F[_]: Async: ContextShift]
-      : Kleisli[F, BlobPropsArgs, BlobGetPropertiesResponse] =
+      : Kleisli[F, BlobPropsArgs, Response[BlobProperties]] =
     Kleisli(blobPropsRequest[F])
 
-  final case class DownloadArgs(
-    blobURL: BlobURL,
-    blobRange: BlobRange,
-    blobAccessConditions: BlobAccessConditions,
-    rangeGetContentMD5: Boolean,
-    context: Context)
+  final case class DownloadArgs(blobClient: BlobAsyncClient)
 
-  def downloadRequest[F[_]: Async: ContextShift](args: DownloadArgs): F[DownloadResponse] =
-    Sync[F].delay(
-      args.blobURL.download(
-        args.blobRange,
-        args.blobAccessConditions,
-        args.rangeGetContentMD5,
-        args.context)) >>=
-      rx.singleToAsync[F, DownloadResponse]
-
-  def downloadRequestK[F[_]: Async: ContextShift]: Kleisli[F, DownloadArgs, DownloadResponse] =
+  def downloadRequest[F[_]: ConcurrentEffect: ContextShift](
+      args: DownloadArgs)
+      : F[Stream[F, ByteBuffer]] =
```

Comment:
Idiomatically, this should be an `F[Stream[F, Byte]]` rather than using `ByteBuffer`. `Chunk.byteBuffer` can be used to achieve this (or `Chunk.byteVector`).